tao_compiler/mlir/disc/transforms/shape_utils.cc

```diff
@@ -656,7 +659,9 @@ SymbolDim* ShapeAnalysis::getRootDim(SymbolDim* symbolDim) {
 SymbolDim* ShapeAnalysis::getDim(Value value, int64_t dim) {
   auto it = shapeMap_.find(value);
   if (it == shapeMap_.end()) return nullptr;
-
+  if (it->second.rank() <= 0 || dim < 0 || dim > it->second.rank() - 1) {
```

Comment:
add a file-check based UT?

Reply:
+1

Reply:
added

Reply:
Thanks. I tried this test out, and disc-opt will crash without the fix of this pr.